### ISSUE TITLE
Add a benchmark for lots of tiny objects to measure latency

### ISSUE THREAD

**Issue by coderedart** [djkoloski/rust_serialization_benchmark#83]
It seems most of the benchmarks are focusing on serializing/deserializing a single piece of huge data. It would be nice to have a benchmark focusing on lots of small data. 

In a lot of real world cases, we deal with lots of small messages (<1KB), rather than huge 1MB+ files. So, It could be a useful benchmark.

**Comment by djkoloski** [comment]
I think this is a good idea. It would also provide some pressure to reduce the overhead involved in performing a single serialization operation.

**Comment by mumbleskates** [comment]
since each of the benchmarks' data is represented as an outer vector of smaller data, this could be accomplished by moving the benchmarked action down one level of nesting. this just omits the cost of whatever bookkeeping is involved in that outer packaging. this could probably be done with relatively little code...

**Comment by louaykamel** [comment]
Agreed, I would if we can have something similar but for primitive types [bench example](https://github.com/louaykamel/wiring/blob/master/wiring/benches/benches.rs).

**Comment by finnbear** [comment]
IMO the real world use case of high-frequency trading would be a good fit:
```
/// One per call to serialize/deserialize.
struct Order {
	sequence_number: u32,
        ticker: String,
	account: String,
	order_type: OrderType,
	side: Side,
	time_in_force: TimeInForce,
}

/// https://www.onixs.biz/fix-dictionary/4.4/tagnum_40.html
enum OrderType {
	Market,
	Limit,
}

/// https://www.onixs.biz/fix-dictionary/4.4/tagnum_54.html
enum Side {
	Buy,
	Sell,
	SellShort,
}

/// https://www.onixs.biz/fix-dictionary/4.4/tagnum_59.html
enum TimeInForce {
	Day,
	GoodTillCancel,
	AtTheOpening,
	ImmediateOrCancel,
}
```
